src/components/Header/Header.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from "react";
+import { UserType } from "@/utils/types/userTypes";
+import { signOut } from "next-auth/react";
+
+interface Props {
+  user?: UserType;
+}
+
+type TabLinks = {
+  [tabName: string]: string;
+};
+
+const Header = (props: Props) => {
+  const userType = props.user || UserType.Public;
+  const tabData: { [key in UserType]: string[] } = {
+    [UserType.Public]: ["Home", "Game Gallery", "Donate"],
+    [UserType.AccountHolder]: ["Home", "Game Gallery", "Donate"],
+    [UserType.Admin]: [
+      "Home",
+      "Game Gallery",
+      "Donate",
+      "Account Management",
+      "Themes and Tags",
+    ],
+  };
+
+  const tabLinks: TabLinks = {
+    Home: "/",
+    "Game Gallery": "/games",
+    Donate: "https://www.paypal.com/paypalme/stopTDV",
+    "Account Management": "/admin/account-management",
+    "Themes and Tags": "/admin/themes",
+  };
+
+  const [selectedTab, setSelectedTab] = useState<number>(0);
+
+  function handlePageChange(tabname: string, index: number) {
+    if (tabname != "Donate") {
+      setSelectedTab(index);
+    }
+  }
+
+  function handleSignUpLogOut() {
+    if (userType != UserType.Public) {
+      signOut();
+    } else {
+      window.location.href = "/signup";
```

Comment:
to improve user experience and align with the rest of the application, could you use
```
import { useRouter } from "next/router";
const router = useRouter();
router.push("/signup");
```